crates/swc_plugin_runner/src/lib.rs

```diff
@@ -9,7 +17,40 @@ pub mod resolve;
 /// Load plugin from specified path.
 /// If cache is provided, it'll try to load from cache first to avoid
 /// compilation.
-fn load_plugin(module_bytes: &Vec<u8>, cache: &mut Option<PluginCache>) -> Result<Instance, Error> {
+///
+/// Since plugin will be initialized per-file transform, this function tries to
+/// avoid reading filesystem per each initialization via naive in-memory map
+/// which stores raw bytecodes from file. Unlike compiled bytecode cache for the
+/// wasm, this is volatile.
+///
+/// ### Notes
+/// [This code](https://github.com/swc-project/swc/blob/fc4c6708f24cda39640fbbfe56123f2f6eeb2474/crates/swc/src/plugin.rs#L19-L44)
+/// includes previous incorrect attempt to workaround file read issues.
+/// In actual transform, `plugins` is also being called per each transform.
+fn load_plugin(plugin_path: &Path, cache: &mut Option<PluginCache>) -> Result<Instance, Error> {
+    static BYTE_CACHE: Lazy<Mutex<AHashMap<PathBuf, Arc<Vec<u8>>>>> =
+        Lazy::new(|| Default::default());
+
+    // TODO: This caching streategy does not consider few edge cases.
+    // 1. If process is long-running (devServer) binary change in the middle of
+    // process won't be reflected.
+    // 2. If reading binary fails somehow it won't bail out but keep retry.
+    let module_bytes_key = plugin_path.to_path_buf();
+    let module_bytes = if let Some(cached_bytes) = BYTE_CACHE.lock().get(&module_bytes_key).cloned()
```

Comment:
I think type of this can be `Arc<Vec<u8>>>` instead of `Result` as the cons and alt produce identical type.

Reply:
yes, it's my oversight.